### PR TITLE
improve rules test coverage for pdo-pgsql

### PR DESCRIPTION
**Pull request by p4veI**
Only query analyzer rule test is skipped for pgsql now, otherwise everything should have coverage with this.

Related to: #371

### CODE REVIEW

**Comment by p4veI** [comment]
@staabm btw I'm still failing to understand how did you manage to make the pgsql test case pass with my previous PR, I'm unable to achieve it locally.

pdo-prepare.php:94 with the integer range still fails. But thanks for pushing it forward, feel free to assign me to anything else, I'll be happy to help and since it might not be such a big chunk of code now, easier to push over the finish line :)

**Comment by staabm** [comment]
> pdo-prepare.php:94 with the integer range still fails.

Could you open a separate issue with the exact error message you see?

> feel free to assign me to anything else, I'll be happy to help 

Really cool, thx

File: tests/rules/SyntaxErrorInQueryFunctionRulePdoReflectorTest.php [hunk 1]
```diff
@@ -46,4 +46,40 @@ public function testSyntaxErrorInQueryRule(): void
             ],
         ]);
     }
+
+    public function testSyntaxErrorInPgsqlQueryRule(): void
+    {
+        if ('pdo-pgsql' !== getenv('DBA_REFLECTOR')) {
+            $this->markTestSkipped('Only works with PdoMysqlQueryReflector');
```
Comment (staabm) [inline tests/rules/SyntaxErrorInQueryFunctionRulePdoReflectorTest.php:53]:
```suggestion
            $this->markTestSkipped('Only works with PdoPgsqlQueryReflector');
```

File: tests/rules/SyntaxErrorInQueryMethodSubclassedRulePdoReflectorTest.php [hunk 2]
```diff
@@ -42,4 +42,32 @@ public function testSyntaxErrorInQueryRule(): void
             ],
         ]);
     }
+
+    public function testSyntaxErrorInPgsqlQueryRule(): void
+    {
+        if ('pdo-pgsql' !== getenv('DBA_REFLECTOR')) {
+            $this->markTestSkipped('Only works with PdoMysqlQueryReflector');
```
Comment (staabm) [inline tests/rules/SyntaxErrorInQueryMethodSubclassedRulePdoReflectorTest.php:49]:
```suggestion
            $this->markTestSkipped('Only works with PdoPgsqlQueryReflector');
```

**Pull request merged**